AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m

```diff
@@ -1140,15 +1160,81 @@ - (void)testOverridingCommonSchemaPropertiesWithTwoChildrenUnderTheSameParent {
   XCTAssertEqualObjects(parentLog.ext.appExt.ver, parent.propertyConfigurator.appVersion);
   XCTAssertEqualObjects(parentLog.ext.appExt.name, parent.propertyConfigurator.appName);
   XCTAssertEqualObjects(parentLog.ext.appExt.locale, parent.propertyConfigurator.appLocale);
+  XCTAssertEqualObjects(parentLog.ext.appExt.userId, parent.propertyConfigurator.appUserId);
   XCTAssertEqualObjects(child1Log.ext.appExt.ver, child1.propertyConfigurator.appVersion);
   XCTAssertEqualObjects(child1Log.ext.appExt.name, child1.propertyConfigurator.appName);
   XCTAssertEqualObjects(child1Log.ext.appExt.locale, child1.propertyConfigurator.appLocale);
+  XCTAssertEqualObjects(child1Log.ext.appExt.userId, child1.propertyConfigurator.appUserId);
   XCTAssertEqualObjects(child2Log.ext.appExt.ver, parent.propertyConfigurator.appVersion);
   XCTAssertEqualObjects(child2Log.ext.appExt.name, parent.propertyConfigurator.appName);
   XCTAssertEqualObjects(child2Log.ext.appExt.locale, parent.propertyConfigurator.appLocale);
+  XCTAssertEqualObjects(child2Log.ext.appExt.userId, parent.propertyConfigurator.appUserId);
   XCTAssertNil(child2.propertyConfigurator.appVersion);
   XCTAssertNil(child2.propertyConfigurator.appName);
   XCTAssertNil(child2.propertyConfigurator.appLocale);
+  XCTAssertNil(child2.propertyConfigurator.appUserId);
+}
+
+- (void)testOverridingInvalidAppUserId {
+
+  // If
+  MSAnalyticsTransmissionTarget *target = [MSAnalytics transmissionTargetForToken:@"invalidUserAppIdTest"];
+
+  // Set invalid user identifier (no prefix).
+  [target.propertyConfigurator setAppUserId:@"invalid"];
+
+  // Set a log.
+  MSCommonSchemaLog *log = [MSCommonSchemaLog new];
+  log.tag = target;
+  [log addTransmissionTargetToken:@"invalidUserAppIdTest"];
+  log.ext = [MSCSExtensions new];
+  log.ext.appExt = [MSAppExtension new];
+
+  // When
+  [target.propertyConfigurator channel:nil prepareLog:log];
+
+  // Then
+  XCTAssertNil(log.ext.appExt.userId);
+}
+
+- (void)testOverridingValidAppUserIdThenUnset {
```

Comment:
In addition to this test case, can we add a test case where overriding with valid AppUserId then overriding with invalid AppUserId to make sure that the second AppUserId doesn't override the first AppUserId?

Reply:
Done in https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1229/commits/30851986d2b5b4679249622a2ad729fe948c0a4c